### PR TITLE
API/resize: fix failure on greyscale images

### PR DESCRIPTION
**Pull request by dfdan**
:heart: Thank you for your contribution!

### Description

As reported by @fenekku - resize operations were failing on greyscale images. This was caused by the method specified in IIIIF_RESIZE_RESAMPLE being applied to all image regardles of mode.

This setting appears to have been added to force Pillow to perform (higher quality) bicubic resampling on colour images. 
Since Pillow 7.0, bicubic is used by default on colour images - thus we can simply bump the Pillow version requirement and remove the override / allow Pillow to select the most appropriate method. 

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [x] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: flask_iiif/api.py [hunk 1]
```diff
@@ -206,6 +201,8 @@ def resize(self, dimensions, resample=None):
             )
 
         arguments = dict(size=(width, height), resample=resample)
+        if resample:
+            arguments["resample"] = resample
```
Comment (fenekku) [inline flask_iiif/api.py:205]: This is not needed given the line above, right? `arguments` will already have a `"resample"` key set to the value passed as argument to the `resize` function now that the `resample` assignment from the setting has been removed.

Comment (dfdan) [reply flask_iiif/api.py:205]: Oops, meant to take it out from the line above - I don't wnt to pass "resample=None" to resize.

Comment (fenekku) [reply flask_iiif/api.py:205]: or better yet:

```python
arguments = {...}  # without resample
if resample is not None:
    arguments["resample"] = resample
```

**Comment by fenekku** [comment]
I can't merge on this repo. Maybe someone else with merging powers?

**Review by slint — APPROVED** [review]

**Pull request merged**